### PR TITLE
Commvault Plugin: port CommvaultOptionSourceService#commvaultStoragePolicies to a Dataset

### PR DESCRIPTION
**Pull request by razi-ahmad1**

### CODE REVIEW

File: src/main/groovy/com/morpheusdata/commvault/datasets/CommvaultStoragePoliciesDatasetProvider.groovy [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+            if(longValue) {
+                rtn = item(longValue)
+            }
+        }
+        return rtn
+    }
+
+    @Override
+    ReferenceData item(Long value) {
+        def rtn = list(new DatasetQuery()).find{ it.id == value }
```
Comment (ddeyoung) [inline src/main/groovy/com/morpheusdata/commvault/datasets/CommvaultStoragePoliciesDatasetProvider.groovy]: This isn't going to return any data, calling `list()` with a blank DatasetQuery. List requires a backupProvider and a cloud to return any data.

item should really just be `return morpheus.services.referenceData.get(value)`.

File: src/main/groovy/com/morpheusdata/commvault/datasets/CommvaultStoragePoliciesDatasetProvider.groovy [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        }
+        if(cloud && cloud.backupProvider) {
+            backupProvider = morpheusContext.services.backupProvider.find(new DataQuery().withFilter("account", cloud.account)
+                    .withFilter("id", cloud.backupProvider.id))
+        }
+
+        if(backupProvider) {
+            def results = morpheusContext.services.referenceData.list(new DataQuery()
+                    .withFilter("category", "${backupProvider?.type?.code}.backup.storagePolicy.${backupProvider?.id}"))
+            if (results.size() > 0) {
```
Comment (ddeyoung) [inline src/main/groovy/com/morpheusdata/commvault/datasets/CommvaultStoragePoliciesDatasetProvider.groovy]: Everything before this should be done in `list()` and then converted to a map in `listOptions()`:
```
list(datasetQuery).toList().blockingGet().collect {refData -> [name: refData.name, id: refData.id, value: refData.id] }
```

**Review by ddeyoung — CHANGES_REQUESTED** [review]

File: src/main/groovy/com/morpheusdata/commvault/datasets/CommvaultStoragePoliciesDatasetProvider.groovy (hunk 1 above)
Comment (razi-ahmad1) [reply src/main/groovy/com/morpheusdata/commvault/datasets/CommvaultStoragePoliciesDatasetProvider.groovy]: @ddeyoung changes has been incorporated as suggested.

File: src/main/groovy/com/morpheusdata/commvault/datasets/CommvaultStoragePoliciesDatasetProvider.groovy (hunk 2 above)
Comment (razi-ahmad1) [reply src/main/groovy/com/morpheusdata/commvault/datasets/CommvaultStoragePoliciesDatasetProvider.groovy]: @ddeyoung changes has been incorporated as suggested.

**Review by ddeyoung — APPROVED** [review]

**Pull request merged**